client/rest-high-level/src/main/java/org/elasticsearch/client/indexlifecycle/SetPriorityAction.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.client.indexlifecycle;
+
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.xcontent.ConstructingObjectParser;
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.ToXContentObject;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+
+import java.io.IOException;
+
+/**
+ * A {@link LifecycleAction} which sets the index's priority. The higher the prior, the faster the recovery.
+ */
+public class SetPriorityAction implements LifecycleAction, ToXContentObject {
+    public static final String NAME = "set_priority";
```

Comment:
What do you think of `set_recovery_priority`? That's a little more verbose, but I think it's also clearer to someone who's not an expert that might be trying to read and understand a policy.

(This is not necessarily a change request, just a thought)

Reply:
I think we should mirror the actual setting since if (for what ever reason), the semantics of `index.priority` changes (i.e. used for more then just recovery), then ILM naming stays true. 

Also, not sure if adding `recovery` wouldn't just change the non-expert's question from what does `priority` mean to what does `recovery_priorty` mean.

Reply:
Personally I think since we only have one priority setting and it's called "index priority" we should stick with the parity there and leave it as "set_priority", but I am interested in what others think